pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubit.hpp

```diff
@@ -632,5 +643,133 @@ class StateVectorLQubit : public StateVectorBase<PrecisionT, Derived> {
 
         applyMatrix(matrix.data(), wires, inverse);
     }
+
+    /**
+     * @brief Apply a mid-circuit measurement.
+     *
+     * @param wires Wires to sample from.
+     */
+    inline auto
+    applyMidMeasureMP(const std::vector<std::size_t> &wires,
+                      const std::vector<std::size_t> &postselect = {},
+                      bool reset = false) -> int {
+        PL_ABORT_IF_NOT(wires.size() == 1,
+                        "MidMeasureMP should have a single wire.")
+        PL_ABORT_IF(postselect.size() > 1,
+                    "MidMeasureMP accepts at most one postselect value.")
+        const std::size_t ps = (postselect.size() == 0) ? -1 : postselect[0];
+        return measure(wires[0], ps, reset);
+    }
+
+    /**
+     * @brief Set the seed of the internal random generator
+     *
+     * @param seed Seed
+     */
+    void seed(const size_t seed) { rng.seed(seed); }
+
+    /**
+     * @brief Sample 0 or 1 for given probabilities
+     *
+     * @param probs Probabilities of 0.
+     */
+    const auto random_sample(const PrecisionT prob_0) {
+        std::discrete_distribution<int> distrib({prob_0, 1. - prob_0});
+        return distrib(rng);
+    }
+
+    /**
+     * @brief Calculate probabilities of measuring 0 or 1 on a specific wire
+     *
+     * @param wire Wire
+     * @return std::vector<PrecisionT> Both probabilities of getting 0 or 1
+     */
+    auto probs(const std::size_t wire) -> std::vector<PrecisionT> {
```

Comment:
Could we remove this in favor of using `probs` from `core/src/simulators/lightning_qubit/measurements/MeasurementsLQubit.hpp`?

Reply:
This introduces a circular dependency between `MeasurementsLQubit.hpp` and `StateVectorLQubit.hpp`. Let's keep it as it is right now, and fix it in another story.

Reply:
We need the marginal probability distribution of the measured wire to get a sample which will determine which state the state vector collapses to